src/hip_memory.cpp

```diff
@@ -1411,6 +1411,65 @@ hipError_t hipMemcpyToArray(hipArray* dst, size_t wOffset, size_t hOffset,
     return ihipLogStatus(e);
 }
 
+hipError_t hipMemcpyFromArray(void* dst, hipArray_const_t srcArray, size_t wOffset, size_t hOffset,
```

Comment:
The implementation of the 3 functions is nearly identical. They could be implemented using a single ihipMemcpyFromArray function that takes ```kind``` with appropriate values being passed for it in the 3 functions.